code/go/0chain.net/blobbercore/filestore/storage.go

```diff
@@ -0,0 +1,628 @@
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		return err
+	}
+
+	size := finfo.Size()
+	err = os.Remove(fileObjectPath)
+	if err != nil {
+		logging.Logger.Warn("invalid_path", zap.String("fileObjectPath", fileObjectPath), zap.Error(err))
+		return err
+	}
+
+	fs.updateAllocTempFileSize(allocID, -size)
+
+	return nil
+}
+
+func (fs *FileStore) GetFileBlock(allocID string, fileData *FileInputData, blockNum, numBlocks int64) ([]byte, error) {
+	if blockNum < 1 {
+		return nil, common.NewError("invalid_block_number", "Invalid block number. Start block number must be greater than 0")
+	}
+
+	fileObjectPath, err := fs.GetPathForFile(allocID, fileData.Hash)
+	if err != nil {
+		return nil, common.NewError("get_file_path_error", err.Error())
+	}
+	file, err := os.Open(fileObjectPath)
+	switch {
+	case err != nil && errors.Is(err, os.ErrNotExist):
+		if fs.mc == nil {
+			return nil, common.NewError("file_exist_error", fmt.Sprintf("%s does not exists", fileObjectPath))
+		}
+		err = fs.MinioDownload(fileData.Hash, fileObjectPath)
+		if err != nil {
+			return nil, common.NewError("minio_download_failed", "Unable to download from minio with err "+err.Error())
+		}
+		file, err = os.Open(fileObjectPath)
+		if err != nil {
+			return nil, err
+		}
+	case err != nil:
+		return nil, err
+	}
+
+	defer file.Close()
+	fileinfo, err := file.Stat()
+	if err != nil {
+		return nil, err
+	}
+
+	filesize := fileinfo.Size()
+	maxBlockNum := int64(math.Ceil(float64(filesize) / float64(fileData.ChunkSize)))
+
+	if blockNum > maxBlockNum {
+		return nil, common.NewError("invalid_block_number",
+			fmt.Sprintf("Invalid block number. Start block %d is greater than maximum blocks %d", blockNum, maxBlockNum))
+	}
+
+	buffer := make([]byte, fileData.ChunkSize*numBlocks)
+	n, err := file.ReadAt(buffer, ((blockNum - 1) * fileData.ChunkSize))
+	if err != nil && err != io.EOF {
+		return nil, err
+	}
+
+	return buffer[:n], nil
+}
+
+func (fs *FileStore) GetFileBlockForChallenge(allocID string,
+	fileData *FileInputData, blockoffset int) (json.RawMessage, util.MerkleTreeI, error) {
+
+	if blockoffset < 0 || blockoffset >= 1024 {
+		return nil, nil, common.NewError("invalid_block_number", "Invalid block offset")
+	}
+
+	fileObjectPath, err := fs.GetPathForFile(allocID, fileData.Hash)
+	if err != nil {
+		return nil, nil, common.NewError("get_file_path_error", err.Error())
+	}
+
+	file, err := os.Open(fileObjectPath)
+	switch {
+	case err != nil && errors.Is(err, os.ErrNotExist):
+		if fs.mc == nil {
+			return nil, nil, common.NewError("file_exist_error", fmt.Sprintf("%s does not exists", fileObjectPath))
+		}
+		err = fs.MinioDownload(fileData.Hash, fileObjectPath)
+		if err != nil {
+			return nil, nil, common.NewError("minio_download_failed", "Unable to download from minio with err "+err.Error())
+		}
+		file, err = os.Open(fileObjectPath)
+		if err != nil {
+			return nil, nil, err
+		}
+	case err != nil:
+		return nil, nil, err
+	}
+	defer file.Close()
```

Comment:
@peterlimg They are not same. Even the purpose of function parameters are different.
I think `GetMerkleTree` would be proper name, right?

Reply:
It's for getting data blocks for challenge, not for MerkleTree calculation purpose, can you change it back to the previous name as its more obvious. Otherwise we may misuse it.